### PR TITLE
Transition from using setup.py to pyproject.toml to specify project metadata

### LINKED ISSUE #763: Transition from using setup.py to pyproject.toml to specify project metadata
### Problem Description
- We currently use **setup.py** and **setup.cfg** to specify the project metadata. This includes requirements which are scatter in `apt.txt`, `requirements.txt`, and `tox.ini`.
- The Python community selected pyproject.toml as a standard way of specifying project metadata, removing the need for all these files. All the project metadata should go in this file. 
- Other modern packaging and dependency management projects, like [Python Poetry](https://python-poetry.org/), only support the more declarative way of using a pyproject.toml file.

### Expected behavior
- Use `pyproject.toml` as the one file for managing project metadata

### PR DESCRIPTION
**Pull request by R-Palazzo**
CU-86azg1ghp
Resolve #763

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86azg1ghp RDT - Transition from using setup.py to pyroject.toml to specify project metadata #763](https://app.clickup.com/t/86azg1ghp)

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+dependencies = [
+    "numpy>=1.20.0,<2;python_version<'3.10'",
+    "numpy>=1.23.3,<2;python_version>='3.10'",
+    "pandas>=1.1.3;python_version<'3.10'",
+    "pandas>=1.3.4;python_version>='3.10' and python_version<'3.11'",
+    "pandas>=1.5.0;python_version>='3.11'",
+    "scipy>=1.5.4,<2;python_version<'3.10'",
+    "scipy>=1.9.2,<2;python_version>='3.10'",
+    "scikit-learn>=0.24,<2;python_version<'3.10'",
+    "scikit-learn>=1.1.3,<2;python_version>='3.10'",
```
Comment (amontanez24) [inline pyproject.toml:35]: I think python 3.10 can support scikit-learn 1.1.0
```suggestion
    "scikit-learn>=0.24,<2;python_version<'3.10'",
    "scikit-learn>=1.1.0;python_version>='3.10' and python_version<'3.11'",
    "scikit-learn>=1.1.3,<2;python_version>='3.11'",
```

File: rdt/__init__.py [hunk 2]
```diff
@@ -94,82 +87,3 @@ def get_demo(num_rows=5):
...
-
-        if isinstance(addon, ModuleType):
-            addon_module_name = f'{addon_target.__name__}.{addon_name}'
-            if addon_module_name not in sys.modules:
-                sys.modules[addon_module_name] = addon
-
-        setattr(addon_target, addon_name, addon)
-
-
-_find_addons()
```
Comment (pvk-developer) [inline rdt/__init__.py:175]: This code has to stay! 😅 Otherwise enterprise addons won't work

**Review by pvk-developer — CHANGES_REQUESTED** [review]

File: rdt/__init__.py (hunk 2 above)
Comment (R-Palazzo) [reply rdt/__init__.py]: Good catch, thanks! Updated in [f332d2d](https://github.com/sdv-dev/RDT/pull/766/commits/f332d2de8b98a4c2e491008b6aed7f629aeeaff2)

**Review by pvk-developer — APPROVED** [review]
LGTM ;)!

**Review by amontanez24 — APPROVED** [review]
LGTM!

**Pull request merged**